### PR TITLE
Fix dict prompt for default value

### LINKED ISSUE #882: Accepting default dict parameter value with no user input causes string "default" to be put into dict parameter
* Cookiecutter version: 1.5
* Template project url: private repo, minimal example is attached
* Python version: 2.7.10
* Operating System: Mac OS X 10.10.5

### Description:

I'm using dict parameter and trying to let it use default value. When prompted, `[default]` is shown as dict parameter default value. Then, if I respond to prompt with:

1. Just enter (i.e. no user input) - project cutting fails due to an error, as cookiecutter puts `default` into dict parameter (i.e. it becomes string instead of dict).
2. String "default" and enter - project is cooked successfully with default value.

In two words, just accepting default value for dict parameter by providing no user input causes string "default" to be put into a dict parameter instead of default value provided in cookiecutter.json.

If you allow some speculation, this happens because `click.prompt` [does not call `value_proc` if no user input is given](https://github.com/pallets/click/blob/master/click/termui.py#L103-L104)

See console session logs below.

### What I've run:

Failure:
```
cookiecutter minimal-example/
app_name [dict_default_bug]: 
dict [default]: 
Unable to create file 'placeholder.md'
Error message: 'str object' has no attribute 'foo'
Context: {
    "cookiecutter": {
        "app_name": "dict_default_bug", 
        "dict": "default"
    }
}
```

Success:
```
cookiecutter minimal-example/
app_name [dict_default_bug]: 
dict [default]: default
```
[minimal-example.zip](https://github.com/audreyr/cookiecutter/files/686416/minimal-example.zip)

### PR DESCRIPTION
**Pull request by hackebrot**
Resolve #882

### CODE REVIEW

**Comment by e-kolpakov** [comment]
@hackebrot Works like a charm!

Just a minor question - any chance to see this on pypi soon? in two words, I'm advocating to use cookiecutter at work; mostly a Java/Scala shop. So, the simpler the setup the better, i.e. they would totally be ok with `brew install pip; pip install cookiecutter==1.5.1`, but `brew install pip; pip install git+https://github.com/.../cookiecutter.git@commit_hash` might see some resistance.

File: tests/test_read_user_dict.py [hunk 1]
```diff
@@ -105,25 +91,39 @@ def test_should_raise_type_error(mocker):
 def test_should_call_prompt_with_value_proc(mocker):
     """Test to make sure that create_value_proc is actually being used
     to generate a processer for the user input."""
```
Comment (michaeljoseph) [inline tests/test_read_user_dict.py]: Minor doc fix:
> Test to make sure that **process_json** is actually being used to **validate** the user input

Comment (hackebrot) [inline tests/test_read_user_dict.py]: Good spot, @michaeljoseph! Updated the test name and doc string.

**Comment by e-kolpakov** [comment]
@hackebrot friendly reminder to push a bugfix release.

**Comment by hackebrot** [comment]
Hi @e-kolpakov! 👋 

Thank you for reminding me. Unfortunately other things require my attention right now. I hope to get this release soon.

**Comment by e-kolpakov** [comment]
@hackebrot Sure, no worries!

**Comment by hackebrot** [comment]
Hey @e-kolpakov! 👋 

I just pushed the hotfix to PyPI. Please let me know in case of any issues.

Thank you for your bug report.

**Comment by e-kolpakov** [comment]
@hackebrot Thanks - works like a charm!

**Pull request merged**